### PR TITLE
Issue #11734: Add codenarc to CI for groovy files 

### LINKED ISSUE #11734: Add codenarc to CI for groovy files
Now that we are keeping groovy files in the main Checkstyle repo(after #11731), we should have a static analysis tool to make sure that we keep the same quality in Groovy code that we do for Java.

>Let's add codenarc to CI execution now that we are adding groovy to main repo. Take a look at contribution to see how it is done there: https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/codenarc.sh

_Originally posted by @nick-mancuso in https://github.com/checkstyle/checkstyle/issues/11731#issuecomment-1153228921_

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #11734

Failing CodeNarc analysis (Proof of working): https://github.com/checkstyle/checkstyle/runs/7021070841?check_suite_focus=true

### CODE REVIEW

**Comment by romani** [comment]
I would move this check to Travis or other not high demand CI. It it most likely that such script will never be updated after summer, so better to have it run in some less critical CI.

But we can do this move at end of summer

File: .ci/StarterRuleSet-AllRulesByCategory.groovy.txt [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    // rulesets/unused.xml
+    UnusedArray
+    UnusedMethodParameter
+    UnusedObject
+    UnusedPrivateField
+    UnusedPrivateMethod
+    UnusedPrivateMethodParameter
+    UnusedVariable
+
+
```
Comment (nick-mancuso) [inline .ci/StarterRuleSet-AllRulesByCategory.groovy.txt:425]: Please remove empty lines

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Items:

File: .ci/StarterRuleSet-AllRulesByCategory.groovy.txt (hunk 1 above)
Comment (Vyom-Yadav) [reply .ci/StarterRuleSet-AllRulesByCategory.groovy.txt]: Done.

File: .ci/jsoref-spellchecker/whitelist.words [hunk 2]
```diff
@@ -1424,6 +1431,7 @@ Weblogic
 Webp
 website
 webtoolkit
+weired
```
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply .ci/jsoref-spellchecker/whitelist.words]: Came from the starter file I copied, updated it with https://codenarc.org/StarterRuleSet-AllRulesByCategory.groovy.txt, done.

[the remark this answers is not in the document]
Comment (nick-mancuso) [reply .ci/jsoref-spellchecker/whitelist.words]: I was OK with all rule suppressions, just not misspellings that came with them :)

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply .ci/jsoref-spellchecker/whitelist.words]: Resolved?

**Comment by nick-mancuso** [comment]
@Vyom-Yadav see CI failure at https://cloud.drone.io/checkstyle/checkstyle/10876/5/2

**Comment by Vyom-Yadav** [comment]
> @Vyom-Yadav see CI failure at https://cloud.drone.io/checkstyle/checkstyle/10876/5/2

Done.

**Review by nick-mancuso — APPROVED** [review]

**Pull request merged**